### PR TITLE
[argtable3] actually disable examples

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?
  Follow up on #23655
  The switch to disable examples didn't exist in the version that vcpkg was using. Therefore, I updated the port to its latest commit on master.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/argtable3/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
-   FILENAME "argtable-v3.2.1.52f24e5.tar.gz"
-   SHA512 cec77d56048b38bb7af8553cb660e745972bbd90378eeea4e928579af78190c8a41fdb29c972263e18955e3a497e09c42f705f7c4d548c3c523c5cb104c97a10
-)
-
-vcpkg_extract_source_archive_ex(
+vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
+    REPO argtable/argtable3
+    REF 7704006f3cbb556e11da80a5b97469075a32892e
```
Comment (strega-nil-ms) [inline ports/argtable3/portfile.cmake:4]:
```suggestion
    REF 7704006f3cbb556e11da80a5b97469075a32892e # v3.2.1 + ARGTABLE3_ENABLE_EXAMPLES option
```

File: ports/argtable3/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "argtable3",
-  "version": "3.2.1",
-  "port-version": 1,
+  "version-date": "2021-06-12",
```
Comment (strega-nil-ms) [inline ports/argtable3/vcpkg.json:3]:
```suggestion
  "version": "3.2.1",
  "port-version": 1,
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
We should not be switching from version to version-date here.

**Review by strega-nil-ms — APPROVED** [review]

**Comment by Thomas1664** [comment]
Sorry, was faster than you

**Pull request merged**